### PR TITLE
Add a `macros` option

### PR DESCRIPTION
**Pull request by phasetr**
I'm glad to have `macros` option.

### CODE REVIEW

File: src/Latex.test.tsx [hunk 1]
```diff
@@ -40,4 +40,11 @@ describe('Latex', () => {
     const latex = "$hello$$world$$boo$";
     const wrapper = shallow(<Latex>{latex}</Latex>);
     expect(wrapper).toMatchSnapshot();
-});
+  });
+
+  it("renders correctly an expression with macros", () => {
```
Comment (harunurhan) [inline src/Latex.test.tsx]:
```suggestion
  it("renders an expression with macros", () => {
```

File: README.md [hunk 2]
```diff
@@ -63,6 +63,12 @@ You can enable strict mode like below, which will throw the error instead.
 <Latex strict>{textWithSomeBrokenLatex}</Latex>
 ```
 
+### macros
+
```
Comment (harunurhan) [inline README.md:67]: Could you add sth like

> A collection of custom macros. Each macro is a property with a name like \name (written "\\name" in JavaScript) which maps to a string that describes the expansion of the macro.

(from katex docs)

File: src/Latex.tsx [hunk 3]
```diff
@@ -19,11 +20,12 @@ export default class Latex extends React.Component<LatexProps> {
       { left: '\\[', right: '\\]', display: true },
     ],
     strict: false,
+    macros: undefined
```
Comment (harunurhan) [inline src/Latex.tsx]: `undefined` default value is redundant.

File: src/Latex.tsx [hunk 4]
```diff
@@ -8,6 +8,7 @@ export interface LatexProps {
   children: string;
   delimiters?: Delimiter[];
   strict?: boolean;
+  macros?: object
```
Comment (harunurhan) [inline src/Latex.tsx]:
```suggestion
  macros?: { [name: string]: [macro: string] } 
```

File: README.md (hunk 2 above)
Comment (harunurhan) [reply README.md:67]: Also we could add a link to the persistent macros, make sure users pass the same object to all Latex react components if they want macro persistence.

File: src/renderLatex.ts [hunk 5]
```diff
@@ -16,7 +16,8 @@ export default function renderLatexInTextAsHTMLString(text: string, delimiters:
       const latex = data[i].data;
       const displayMode = data[i].display;
       try {
-        const rendered = renderToString(latex, { displayMode });
+        const options = macros ? { displayMode, macros } : { displayMode }
```
Comment (harunurhan) [inline src/renderLatex.ts]: can't you just pass `{ displayMode, macros }`, does `katex` throw if `macros: undefined` is passed?

File: src/Latex.test.tsx (hunk 1 above)
Comment (phasetr) [reply src/Latex.test.tsx]: I fixed it

File: README.md (hunk 2 above)
Comment (phasetr) [reply README.md:67]: I fixed it.

File: src/Latex.tsx (hunk 3 above)
Comment (phasetr) [reply src/Latex.tsx]: I deleted it.

File: src/renderLatex.ts (hunk 5 above)
Comment (phasetr) [reply src/renderLatex.ts]: The code `const rendered = renderToString(latex, { displayMode, macros });` gives an error, 
>semantic error TS2345 Argument of type '{ displayMode: boolean | undefined; macros: object | undefined; }' is not assignable to parameter of type 'KatexOptions | undefined'.
  Object literal may only specify known properties, and 'macros' does not exist in type 'KatexOptions | undefined'.

I am a typescript newbie, how do I resolve it?

File: README.md [hunk 6]
```diff
@@ -63,6 +63,17 @@ You can enable strict mode like below, which will throw the error instead.
...
+A collection of custom macros.
+Each macro is a property with a name like `\name` (written `\\name` in JavaScript) which maps to a string that describes the expansion of the macro.
+Also we could add a link to the persistent macros,
+make sure users pass the same object to all Latex react components if they want macro persistence.
```
Comment (harunurhan) [inline README.md]: ?

**Comment by harunurhan** [comment]
@phasetr thanks for starting the work, fix the remaining issues and merged.

**Comment by phasetr** [comment]
@harunurhan Thank you for your review and merge, too!

**Pull request merged**